### PR TITLE
#126: Add local notification support

### PR DESCRIPTION
**Pull request by NathHorrigan**
This PR add's support for local notifications. These notifications are used for 3 scenarios:
- When a break period ends
- 2 Hours before the end of a break, if the break was over 4 hours.
- If the device was offline when the experiment finished and data wasn't therfore synced, we schedule a notification 2 & 24 hours later.

I've also added a permission screen after the T&C's that allows the Participant to agree/disagree to the use of notifications. Unfortunately for Android support we had to add Firebase as a dependency.

### Reminder to myself:
- Add changes to Tech spec
- Add Firebase Key to Pipeline

### CODE REVIEW

File: src/utils/notifications.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    popInitialNotification: true,
+    requestPermissions: false,
+  })
+
+  // Create the channel for Android
+  PushNotification.createChannel(
+    {
+      channelId: 'flare',
+      channelName: 'FLARe Research',
+      channelDescription: 'A channel to categorise your notifications',
```
Comment (mixxorz) [inline src/utils/notifications.ts]: Placeholder channel description?

File: src/screens/SummaryScreen.tsx [hunk 2]
```diff
@@ -21,7 +25,7 @@ export const SummaryScreen: React.FC<SummaryScreenProps> = ({
   const [isSyncing, setIsSyncing] = useState<boolean>(false)
   const netInfo = useNetInfo()
   const allModulesSynced =
-    modules.filter((modules) => !modules.moduleCompleted).length === 0
+    modules.filter((modules) => modules.moduleSynced == false).length === 0
```
Comment (mixxorz) [inline src/screens/SummaryScreen.tsx]: I'm not sure with Typescript but should we be using `===` here?

Comment (NathHorrigan) [reply src/screens/SummaryScreen.tsx]: No harm in changing, Fixed.

File: src/utils/notifications.ts (hunk 1 above)
Comment (NathHorrigan) [reply src/utils/notifications.ts]: No harm in changing, Updated.

**Pull request merged**